core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -93,16 +94,38 @@ private static StaticDataTask.Row convertPartition(Partition partition) {
     return StaticDataTask.Row.of(partition.key, partition.recordCount, partition.fileCount, partition.specId);
   }
 
-  private static Iterable<Partition> partitions(StaticTableScan scan) {
+  private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
     CloseableIterable<FileScanTask> tasks = planFiles(scan);
 
-    PartitionMap partitions = new PartitionMap(scan.table().spec().partitionType());
+    PartitionMap partitions = new PartitionMap(Partitioning.partitionType(table));
     for (FileScanTask task : tasks) {
-      partitions.get(task.file().partition()).update(task.file());
+      PartitionData original = (PartitionData) task.file().partition();
+      PartitionData normalized = normalizePartition(original, Partitioning.partitionType(table));
+      partitions.get(normalized).update(task.file());
     }
     return partitions.all();
   }
 
+  private static PartitionData normalizePartition(PartitionData partition, Types.StructType newSchema) {
+    Preconditions.checkArgument(partition.getPartitionType().fields().size() == partition.size(),
+        "Partition values must match size");
+    Map<Integer, Object> fieldIdToValues = Maps.newHashMap();
+    int originalPartitionIndex = 0;
+    for (Types.NestedField f : partition.getPartitionType().fields()) {
+      fieldIdToValues.put(f.fieldId(), partition.get(originalPartitionIndex));
+      originalPartitionIndex++;
+    }
+
+    PartitionData result = new PartitionData(newSchema);
```

Comment:
result -> normalizedPartition?

Reply:
Good point, renamed relevant fields in this method.